App.tsx

```diff
@@ -1,14 +1,23 @@
-// App.tsx or index.js
 import React from 'react';
-import { SafeAreaView } from 'react-native';
+import { NavigationContainer } from '@react-navigation/native';
+import { createStackNavigator } from '@react-navigation/stack';
 import PhotoGalleryScreen from './PhotoGalleryScreen';
+import ImageDetailsScreen from './ImageDetailsScreen';
+import ImageModalScreen from './ImageModalScreen'; // Import the ImageModalScreen
+import { RootStackParamList } from './types';
+
+const Stack = createStackNavigator<RootStackParamList>();
 
 const App = () => {
   return (
-    <SafeAreaView style={{ flex: 1 }}>
-      <PhotoGalleryScreen />
-    </SafeAreaView>
+    <NavigationContainer>
+      <Stack.Navigator initialRouteName="PhotoGallery">
+        <Stack.Screen name="PhotoGallery" component={PhotoGalleryScreen} />
+        <Stack.Screen name="ImageDetail" component={ImageDetailsScreen} />
```

Comment:
`ImageDetail` should have its header set to the image URL. This was one of the pieces that we didn't cover in class and I wanted students to figure it out. There are a couple of ways to do it.

1. https://reactnavigation.org/docs/headers/
2. Or setting your navigation options on the screen https://reactnavigation.org/docs/navigation-prop/#setoptions